src/StrawberryShake/Client/src/Transport.WebSockets/DependencyInjection/WebSocketClientBuilderExtensions.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace StrawberryShake.Transport.WebSockets
+{
+    /// <summary>
+    /// Extension methods for configuring an <see cref="IWebSocketClientBuilder"/>
+    /// </summary>
+    public static class WebSocketClientBuilderExtensions
+    {
+        /// <summary>
+        /// Adds a websocket <see cref="ISocketProtocol"/> to the <see cref="WebSocketClient"/>
+        /// </summary>
+        /// <param name="builder">
+        /// The <see cref="IServiceCollection"/>.
+        /// </param>
+        /// <param name="protocol">
+        /// A instance of a protocol to use
+        /// </param>
+        /// <returns>
+        /// An <see cref="IWebSocketClientBuilder"/> that can be used to configure the client.
+        /// </returns>
+        public static IWebSocketClientBuilder AddProtocol(
+            this IWebSocketClientBuilder builder,
+            ISocketProtocol protocol)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            if (protocol == null)
+            {
+                throw new ArgumentNullException(nameof(protocol));
+            }
+
+            builder.Services.Configure<SocketClientFactoryOptions>(
+                builder.Name,
+                options => options.Protocols.Add(protocol));
+
```

Comment:
The protocols are added onto the SocketClientFactoryOptions to later be injected into the socketclient